api/app.py

```
@@ -7,19 +8,150 @@
+        recipe_data['ingredients'] = [
+            {
+                'name': ing.name,
+                'unit': ing.unit,
+                'quantity': str(ing.quantity)
+            } for ing in recipe.ingredients
+        ]
+
+        # Add categories manually
+        recipe_data['categories'] = [cat.name for cat in recipe.categories]
+
+        recipes.append(recipe_data)
+
+    return jsonify(recipes)
+
+
+@app.route('/api/recipes/<int:recipe_id>', methods=['GET'] )
+def get_recipe(recipe_id):
+    recipe = db.session.query(Recipe).options(
+        joinedload(Recipe.ingredients),
+        joinedload(Recipe.categories)
+    ).filter_by(id=recipe_id).first()
+
+    if recipe is None:
+        abort(404, description="Recipe not found")
+
+    recipe_data = recipe.as_dict()
+    recipe_data['pictures'] = recipe_data['pictures'].split(',')
+
+    recipe_data['ingredients'] = [
+        {
+            'name': ing.name,
+            'unit': ing.unit,
+            'quantity': str(ing.quantity)
+        } for ing in recipe.ingredients
+    ]
+
+    recipe_data['categories'] = [cat.name for cat in recipe.categories]
+
+    return jsonify(recipe_data)
+
+
+@app.route('/api/recipes', methods=['POST'])
+def create_recipe():
+    data = request.get_json()
+    new_recipe = Recipe(
+        name=data['name'],
+        pictures=data['pictures'].split(','),
+        instructions=data['instructions'],
+        duration=data['duration'],
+
+    )
+    db.session.add(new_recipe)
+    db.session.flush()
+    for ing in data['ingredients']:
+        new_ingredient = Ingredient(
+            name=ing['name'],
+            unit=ing['unit'],
+            quantity=ing['quantity'],
+            recipe_id=new_recipe.id,
+
+        )
+        db.session.add(new_ingredient)
+
+    for cat in data['categories']:
+        category = db.session.query(Category).filter_by(name=cat).first()
+        if category is None:
+            category = Category(name=cat,color='default')
```

There should be a hex code for a color instead of `'default'`. Also, we don't need to create a `category` that doesn't exist here; we'll use the `create_category()` endpoint. Here you can return an error message if the category does not exist, for example:
```python
return jsonify({'error': f'category {cat["name"]} not found'}), 400
``` 
Same for `edit_recipe()`.